### PR TITLE
Update plugins.md with semver range support specification

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Add support for specifying 'dependencies' for a plugin in its plugin-descriptor.properties file. This property allows for specification of a range of OpenSearch versions for plugin compatibility check.

### Description
_Describe what this change achieves._

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/6433

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
Thanks for this submission, @abseth-amzn! Can you please address the errors? If you have any questions about anything, please let us know.

File: _install-and-configure/plugins.md [hunk 1]
```diff
@@ -247,8 +247,24 @@ bin/opensearch-plugin install --batch <plugin-name>
 
 ## Available plugins
 
-Major, minor, and patch plugin versions must match OpenSearch major, minor, and patch versions in order to be compatible. For example, plugins versions 2.3.0.x work only with OpenSearch 2.3.0.
-{: .warning}
+### Plugin Compatibility
```
Comment (kolchfa-aws) [inline _install-and-configure/plugins.md:250]:
```suggestion
OpenSearch provides several bundled plugins and additional plugins.

### Plugin compatibility
```

File: _install-and-configure/plugins.md [hunk 2]
```diff
@@ -247,8 +247,24 @@ bin/opensearch-plugin install --batch <plugin-name>
...
+
+A plugin can specify explicit compatibility with a specific OpenSearch version. For example, a plugin with following property in its plugin-descriptor.properties is compatible only with OpenSearch 2.3.0
```
Comment (kolchfa-aws) [inline _install-and-configure/plugins.md:252]:
```suggestion
A plugin can explicitly specify compatibility with a specific OpenSearch version by listing that version in its `plugin-descriptor.properties` file. For example, a plugin with the following property is compatible only with OpenSearch 2.3.0:

```

File: _install-and-configure/plugins.md [hunk 3]
```diff
@@ -247,8 +247,24 @@ bin/opensearch-plugin install --batch <plugin-name>
...
+```bash
```
Comment (kolchfa-aws) [inline _install-and-configure/plugins.md:253]:
````suggestion
```properties
````

File: _install-and-configure/plugins.md [hunk 4]
```diff
@@ -247,8 +247,24 @@ bin/opensearch-plugin install --batch <plugin-name>
...
+opensearch.version=2.3.0
+```
+Alternately, a plugin can specify a range of compatible OpenSearch versions using the 'dependencies' property in its plugin-descriptor.properties file. Following notations are supported:
```
Comment (kolchfa-aws) [inline _install-and-configure/plugins.md:256]:
```suggestion
Alternatively, a plugin can specify a range of compatible OpenSearch versions by setting the `dependencies` property in its `plugin-descriptor.properties file` in one of the following notations:
```

File: _install-and-configure/plugins.md [hunk 5]
```diff
@@ -247,8 +247,24 @@ bin/opensearch-plugin install --batch <plugin-name>
...
+- `dependencies={ opensearch: "2.3.0" }`
```
Comment (kolchfa-aws) [inline _install-and-configure/plugins.md:257]:
```suggestion
- `dependencies={ opensearch: "2.3.0" }`: The plugin is compatible only with OpenSearch version 2.3.0.
```

File: _install-and-configure/plugins.md [hunk 6]
```diff
@@ -247,8 +247,24 @@ bin/opensearch-plugin install --batch <plugin-name>
...
+  Plugin is compatible only with OpenSearch version 2.3.0.
```
Comment (kolchfa-aws) [inline _install-and-configure/plugins.md:258]:
```suggestion
```

File: _install-and-configure/plugins.md [hunk 7]
```diff
@@ -247,8 +247,24 @@ bin/opensearch-plugin install --batch <plugin-name>
...
+- `dependencies={ opensearch: "=2.3.0" }`
```
Comment (kolchfa-aws) [inline _install-and-configure/plugins.md:259]:
```suggestion
- `dependencies={ opensearch: "=2.3.0" }`: The plugin is compatible only with OpenSearch version 2.3.0.
```

File: _install-and-configure/plugins.md [hunk 8]
```diff
@@ -247,8 +247,24 @@ bin/opensearch-plugin install --batch <plugin-name>
...
+  Plugin is compatible only with OpenSearch version 2.3.0.
```
Comment (kolchfa-aws) [inline _install-and-configure/plugins.md:260]:
```suggestion
```

File: _install-and-configure/plugins.md [hunk 9]
```diff
@@ -247,8 +247,24 @@ bin/opensearch-plugin install --batch <plugin-name>
...
+- `dependencies={ opensearch: "~2.3.0" }`
```
Comment (kolchfa-aws) [inline _install-and-configure/plugins.md:261]:
```suggestion
- `dependencies={ opensearch: "~2.3.0" }`: The plugin is compatible with all versions starting from 2.3.0 up to the next minor version, in this example, 2.4.0 (exclusive).
```

File: _install-and-configure/plugins.md [hunk 10]
```diff
@@ -247,8 +247,24 @@ bin/opensearch-plugin install --batch <plugin-name>
...
+  Plugin is compatible with all versions starting from 2.3.0 upto next minor version, i.e. 2.4.0 (exclusive).
```
Comment (kolchfa-aws) [inline _install-and-configure/plugins.md:262]:
```suggestion
```

File: _install-and-configure/plugins.md [hunk 11]
```diff
@@ -247,8 +247,24 @@ bin/opensearch-plugin install --batch <plugin-name>
...
+- `dependencies={ opensearch: "^2.3.0" }`
```
Comment (kolchfa-aws) [inline _install-and-configure/plugins.md:263]:
```suggestion
- `dependencies={ opensearch: "^2.3.0" }`: The plugin is compatible with all versions starting from 2.3.0 up to the next major version, in this example, 3.0.0 (exclusive).
```

File: _install-and-configure/plugins.md [hunk 12]
```diff
@@ -247,8 +247,24 @@ bin/opensearch-plugin install --batch <plugin-name>
...
+  Plugin is compatible with all versions starting from 2.3.0 upto next major version, i.e. 3.0.0 (exclusive).
```
Comment (kolchfa-aws) [inline _install-and-configure/plugins.md:264]:
```suggestion
```

File: _install-and-configure/plugins.md [hunk 13]
```diff
@@ -247,8 +247,24 @@ bin/opensearch-plugin install --batch <plugin-name>
...
+
+Only one of `opensearch.version` or `dependencies` properties is allowed to be specified.
```
Comment (kolchfa-aws) [inline _install-and-configure/plugins.md:266]:
```suggestion
You can only specify one of `opensearch.version` or `dependencies` properties.
```

**Comment by abseth-amzn** [comment]
> Thanks for this submission, @abseth-amzn! Can you please address the errors? If you have any questions about anything, please let us know.

Thanks for the suggestions! I have accepted all of them.

File: _install-and-configure/plugins.md [hunk 14]
```diff
@@ -247,8 +247,23 @@ bin/opensearch-plugin install --batch <plugin-name>
...
+```properties
+opensearch.version=2.3.0
+```
+Alternatively, a plugin can specify a range of compatible OpenSearch versions by setting the `dependencies` property in its `plugin-descriptor.properties` file in one of the following notations:
+- `dependencies={ opensearch: "2.3.0" }`: The plugin is compatible only with OpenSearch version 2.3.0.
+- `dependencies={ opensearch: "=2.3.0" }`: The plugin is compatible only with OpenSearch version 2.3.0.
+- `dependencies={ opensearch: "~2.3.0" }`: The plugin is compatible with all versions starting from 2.3.0 up to the next minor version, in this example, 2.4.0 (exclusive).
+- `dependencies={ opensearch: "^2.3.0" }`: The plugin is compatible with all versions starting from 2.3.0 up to the next major version, in this example, 3.0.0 (exclusive).
+
+You can only specify one of `opensearch.version` or `dependencies` properties.
```
Comment (natebower) [inline _install-and-configure/plugins.md:265]:
```suggestion
You can specify only one of the `opensearch.version` or `dependencies` properties.
```

**Review by natebower — APPROVED** [review]
@kolchfa-aws Just a few minor changes. Thanks!

File: _install-and-configure/plugins.md [hunk 15]
```diff
@@ -247,8 +247,23 @@ bin/opensearch-plugin install --batch <plugin-name>
 
 ## Available plugins
 
-Major, minor, and patch plugin versions must match OpenSearch major, minor, and patch versions in order to be compatible. For example, plugins versions 2.3.0.x work only with OpenSearch 2.3.0.
-{: .warning}
+OpenSearch provides several bundled plugins and additional plugins.
```
Comment (kolchfa-aws) [inline _install-and-configure/plugins.md:250]:
```suggestion
OpenSearch provides several bundled and additional plugins.
```

**Comment by kolchfa-aws** [comment]
@abseth-amzn Please accept editorial suggestions, and we'll be ready to merge this PR. Thanks!

**Comment by abseth-amzn** [comment]
I have accepted all the suggestions.

File: _install-and-configure/plugins.md [hunk 16]
```diff
@@ -247,8 +247,23 @@ bin/opensearch-plugin install --batch <plugin-name>
...
+OpenSearch provides several bundled and additional plugins.
+
+### Plugin compatibility
+
+A plugin can explicitly specify compatibility with a specific OpenSearch version by listing that version in its `plugin-descriptor.properties` file. For example, a plugin with the following property is compatible only with OpenSearch 2.3.0:
+
+```properties
+opensearch.version=2.3.0
+```
+Alternatively, a plugin can specify a range of compatible OpenSearch versions by setting the `dependencies` property in its `plugin-descriptor.properties` file using one of the following notations:
```
Comment (rursprung) [inline _install-and-configure/plugins.md:259]: shouldn't you specify that this is only available starting with 2.13.0? (all your examples currently use 2.3.0 where it wouldn't even work)

**Pull request merged**